lib/Runtime/Library/WasmLibrary.cpp

```diff
@@ -106,49 +106,72 @@ Js::JavascriptMethod Js::WasmLibrary::WasmDeferredParseEntryPoint(Js::AsmJsScrip
+            {
+                GenerateFunction(scriptContext->GetNativeCodeGenerator(), body, func);
+                body->SetIsAsmJsFullJitScheduled(true);
+            }
+#endif
+        }
+        catch (Wasm::WasmCompilationException& ex)
         {
-            GenerateFunction(scriptContext->GetNativeCodeGenerator(), body, func);
-            body->SetIsAsmJsFullJitScheduled(true);
+            char16* originalMessage = ex.ReleaseErrorMessage();
+            intptr_t offset = readerInfo->m_module->GetReader()->GetCurrentOffset();
+            intptr_t start = readerInfo->m_funcInfo->m_readerInfo.startOffset;
+            uint32 size = readerInfo->m_funcInfo->m_readerInfo.size;
+
+            Wasm::WasmCompilationException newEx = Wasm::WasmCompilationException(
+                _u("function %s at offset %d/%d: %s"),
+                body->GetDisplayName(),
+                offset - start,
+                size,
+                originalMessage
+            );
+            SysFreeString(originalMessage);
+            char16* msg = newEx.ReleaseErrorMessage();
+            JavascriptLibrary *library = scriptContext->GetLibrary();
+            JavascriptError *pError = library->CreateWebAssemblyCompileError();
+            JavascriptError::SetErrorMessage(pError, WASMERR_WasmCompileError, msg, scriptContext);
+
+            func->GetDynamicType()->SetEntryPoint(WasmLazyTrapCallback);
+            entrypointInfo->jsMethod = WasmLazyTrapCallback;
+            info->SetLazyError(pError);
         }
-#endif
     }
-    catch (Wasm::WasmCompilationException& ex)
+    else
     {
-        char16* originalMessage = ex.ReleaseErrorMessage();
-        intptr_t offset = readerInfo->m_module->GetReader()->GetCurrentOffset();
-        intptr_t start = readerInfo->m_funcInfo->m_readerInfo.startOffset;
-        uint32 size = readerInfo->m_funcInfo->m_readerInfo.size;
-
-        Wasm::WasmCompilationException newEx = Wasm::WasmCompilationException(
-            _u("function %s at offset %d/%d: %s"),
-            body->GetDisplayName(),
-            offset - start,
-            size,
-            originalMessage
-        );
-        SysFreeString(originalMessage);
-        char16* msg = newEx.ReleaseErrorMessage();
-        JavascriptLibrary *library = scriptContext->GetLibrary();
-        JavascriptError *pError = library->CreateWebAssemblyCompileError();
-        JavascriptError::SetErrorMessage(pError, WASMERR_WasmCompileError, msg, scriptContext);
-
-        func->GetDynamicType()->SetEntryPoint(WasmLazyTrapCallback);
-        entypointInfo->jsMethod = WasmLazyTrapCallback;
-        info->SetLazyError(pError);
+        // This can happen if another function had its type changed and then was parsed
+        // They still share the function body, so just change the entry point
+        Assert(body->GetByteCodeCount() > 0);
+        Js::JavascriptMethod externalEntryPoint = info->GetLazyError() ? WasmLazyTrapCallback : Js::AsmJsExternalEntryPoint;
```

Comment:
internal and external calls may have different calling convention (on x86 for instance), so I think in general external we should avoid setting both internal and external entrypoint to same function. (i think in this case it works because we always throw). imo we should keep external as AsmJsExternalEntryPoint.
i think we may also have an issue here in case of with CrossSite, but not totally sure. and if we do, seems really hard to deal with. maybe for release/1.4 we just leave that since i suspect a holistic will be too invasive